spec/api/v4/call_results_spec.rb

```diff
@@ -0,0 +1,38 @@
+require "swagger_helper"
+
+describe "CallResults v4 API", swagger_doc: "v4/swagger.json" do
+  path "/call_results/sync" do
+    post "Syncs call_result data from device to server." do
+      tags "Call Result"
+      security [access_token: [], user_id: [], facility_id: []]
+      parameter name: "HTTP_X_USER_ID", in: :header, type: :uuid
+      parameter name: "HTTP_X_FACILITY_ID", in: :header, type: :uuid
+      parameter name: :call_results, in: :body, schema: Api::V4::Schema.call_result_sync_from_user_request
+
+      response "200", "call_results created" do
+        let(:request_user) { FactoryBot.create(:user) }
+        let(:request_facility) { FactoryBot.create(:facility, facility_group: request_user.facility.facility_group) }
+        let(:HTTP_X_USER_ID) { request_user.id }
+        let(:HTTP_X_FACILITY_ID) { request_facility.id }
+        let(:Authorization) { "Bearer #{request_user.access_token}" }
+
+        let(:call_results) { {call_results: (1..3).map { build_call_result_payload }} }
+
+        run_test!
```

Comment:
shared specs really obscure what is actually being tested here...where would I look to see what is actually done in the test and what assertions are made ?

Reply:
Agree about the shared specs, they really get in the way here. However, since they enforce the shared sync api contracts, I don't feel comfortable not using them for just this API. Maybe we can make a wholesale switch away from them at some point, or find a different way to test the shared contract.

You can find the specific specs included by looking them up in: [shared_spec_for_sync_controller.rb](https://github.com/simpledotorg/simple-server/blob/master/spec/controllers/shared_examples/shared_spec_for_sync_controller.rb).

I managed to split up `sync_from_user` and `sync_to_user` specs so we can just test `sync_from_user` for this endpoint.